perspective_reddit_bot/moderate_subreddit.py

```diff
@@ -199,49 +245,40 @@ def score_subreddit(creds_dict,
   perspective = perspective_client.PerspectiveClient(
     creds_dict['perspective_api_key'])
 
+  current_file_time = datetime.utcnow().strftime('%Y%m%d_%H%M%S')
   for i, comment in enumerate(subreddit.stream.comments()):
     try:
-      # TODO(nthain): Make this loop more readable by breaking into functions.
       if i % 100 == 0 and i > 0:
         print(i)
         # Check if still has mod permissions every 100 comments
         mod_permissions = bot_is_mod(reddit, subreddit)
 
-      original_comment = comment.body
-      comment_for_scoring = (remove_quotes(original_comment)
-                             if should_remove_quotes else original_comment)
-      if len(comment_for_scoring) > 20000:
-        print('Comment too long, skipping...')
+      # Score current comment with perspective models
+      comment_for_scoring, scores = score_comment(comment,
+                                                  perspective,
+                                                  api_models,
+                                                  ensembles,
+                                                  should_remove_quotes)
+      if scores is None:
         continue
-      scores = perspective.score_text(comment_for_scoring, api_models,
-                                      language=LANGUAGE)
-      for e in ensembles:
-        scores[e.name] = e.predict_one(scores)
 
-      action_dict = defaultdict(list)
-      for rule in rules:
-        if rule.check_model_rules(scores):
-          action_dict[rule.action_name].append(rule.rule_description)
-          print_moderation_decision(i, comment, rule)
+      # Check which rules should be applied
+      action_dict = check_rules(i, comment, rules, scores)
 
+      # Apply actions from triggered rules
       if mod_permissions and original_mod_permissions:
```

Comment:
Hm, why is there an 'and' check here?

Also, it seems like `mod_permissions` isn't defined except for every hundredth comment? I'm probably reading this incorrectly right?

Reply:
mod_permissions is defined outside of the loop (~ line 237).
The point of this is twofold:
-  If mod-permissions changes, then the bot will eventually stop taking actions, allowing subreddit moderators to turn the bot off.
- The 'and' with original_mod_permissions prevents the bot from being turned on accidentally. I was thinking of the scenario of the bot being launched for testing and somehow the account getting added as a mod to a new subreddit partway through and it sending them many reporting messages. This 'and' means that reports can only happen if the bot had mod permissions at the time of launch and still has them +/- 100 comments.